### DISCUSSION TITLE
Translate to/from other templating languages

### DISCUSSION THREAD

**Discussion by AntoineThebaud** [cue-lang/cue#1763] [Ideas] [answered]
Hello, I wasn't able to find a clear answer to this thus asking here*: Could it be considered to have the possibility to import/export CUE definitions from/to other templating languages like Helm, Jsonnet, HCL etc.. ?

I think this is purely a matter of improving the adoption of the CUE technology here:
- Converting from another templating language would definitely help to migrate out of it. _Like when importing definitions from Go structs, you'd certainly have to refine the generated code, at least to add some validation that may not exist at all for some languages._
- Converting to another templating language would help in the case you interact with a 3rd party tool that supports this given language but not CUE (yet).

Could this be a thing, or is it already stamped as a no-go?

*related to this topic:
- https://github.com/cue-lang/cue/discussions/1159

**Comment by verdverm** [comment, marked as answer]
With regards to Helm, as an example of the complexities

1. you'd have to parse and work with Go's `text/template` AST
2. There are a number of dynamic values and custom functions
3. I think they support JSONSchema for the values file now, so that might be another option to try
4. Writing a CUE plugin for Helm might be a path for adoption there
5. We'd probably want an "official" k8s import that everyone imports rather than generates, it needs some customization for things like Port (which doesn't have validation because of custom (en/de)coders because it can be either a string or an int)

**Generally, I'd expect that the ecosystem can produce tools like this, rather than putting everything into the cue tool and projects**
